src/golang/cmd/migrator/versions/000016_add_artifact_type_column_to_artifact/utils.go

```diff
@@ -325,6 +326,35 @@ func migrateArtifact(ctx context.Context, db database.Database) error {
 			if err != nil {
 				return err
 			}
+		} else {
+			// If we reach here, it means the artifact has no result available, so we do a best-effort
+			// mapping netween the original artifact type and the new type.
+			if artifactSpec.Spec.spec.Type == TableType {
+				err = updateTypeInArtifact(ctx, artifactSpec.Id, "table", db)
+				if err != nil {
+					return err
+				}
+			} else if artifactSpec.Spec.spec.Type == FloatType {
```

Comment:
any reason we can't add a `artifactSpec.Spec.Type()` convenience function here?  The `Spec.spec` pattern is a bit ugly.